core/tauri/src/plugin.rs

```diff
@@ -60,6 +60,76 @@ type OnWebviewReady<R> = dyn Fn(Window<R>) + Send + Sync;
 type OnEvent<R> = dyn Fn(&AppHandle<R>, &RunEvent) + Send + Sync;
 
 /// Builds a [`TauriPlugin`].
+///
+/// This Builder offers a more concise way to construct Tauri plugins than implementing the Plugin trait directly.
+///
+/// # Conventions
+///
+/// When using the Builder Pattern it is encouraged to export a function called `init` that constructs and returns the plugin.
+/// While plugin authors can provide every possible way to construct a plugin,
+/// sticking to the `init` function convention helps users to quickly identify the correct function to call.
+///
+/// ```rust
+/// use tauri::{plugin::{Builder, TauriPlugin}, runtime::Runtime};
+///
+/// pub fn init<R: Runtime>() -> TauriPlugin<R> {
+///   Builder::new("example")
+///     .build()
+/// }
+/// ```
+///
+/// When plugins expose more complex configuration options, it can be helpful to provide a Builder instead:
+///
+/// ```rust
+/// use tauri::{plugin::{Builder as PluginBuilder, TauriPlugin}, runtime::Runtime};
+///
+/// pub struct Builder {
+///   optionA: String,
```

Comment:
use the rust convention here, option_a instead of optionA

Reply:
Thanks clippy 😉

Reply:
you forgot this part LOL i'm really clippy

Reply:
fixed!